cmd/agent/app/reporter/grpc/tags_merger.go

```diff
@@ -0,0 +1,55 @@
+package grpc
+
+import (
+	"github.com/jaegertracing/jaeger/cmd/agent/app/reporter"
+	"github.com/jaegertracing/jaeger/model"
+)
+
+type TagsMerger interface {
+	// Merge merges jaeger tags for the agent to every span it sends to the collector based on a policy
+	Merge(spans []*model.Span, process *model.Process) ([]*model.Span, *model.Process)
+}
+
+type tagsMerger struct {
+	agentTags           []model.KeyValue
+	duplicateTagsPolicy string
+}
+
+func NewTagsMerger(agentTags []model.KeyValue, duplicateTagsPolicy string) TagsMerger {
```

Comment:
Passing string arg so deep creates unnecessary coupling. We can simply have two constructors, `SimpleTagsMerger(agentTags []model.KeyValue)` and `DedupingTagsMerger(agentTags []model.KeyValue)`

Reply:
As there are three ways that this can be done i.e `Duplcate`, `Client`, `Agent`, we may need three different constructurs and which would be initialised based on the TagPolicy here https://github.com/Pothulapati/jaeger/blob/427afd0328ab82acdf9c6b5a2f602ba2bb3047fe/cmd/agent/app/reporter/grpc/collector_proxy.go#L48

Right?

Reply:
yes